### PR TITLE
Disable voyager when running on Saturn node

### PR DESCRIPTION
**Pull request by juliangruber**
As requested in https://filecoinproject.slack.com/archives/C06DF14LGUA/p1709071237385749

### CODE REVIEW

File: voyager.js [hunk 1]
```diff
@@ -45,31 +45,64 @@ new BasicTracerProvider().register()
...
+    const res = await fetch(`https://api.filspark.com/inspect-request`)
+    const { cloudfareAddr: ip } = await res.json()
+    return ip
+}
+
+async function isRunningOnSaturnNode () {
+    const ip = await getPublicIPv4Address()
+    const subdomain = ip.replaceAll('.', '-')
+    try {
+        await fetch(`https://${subdomain}.l1s.saturn.ms/`, {
```
Comment (travis) [inline voyager.js:101]: I'm pretty sure this is enough, at least for now, but I want @hannahhoward to double check my reasoning that if this returns anything at all there is a Saturn node running on the given IP - couldn't think of any cases where that wouldn't be true (aside from maybe some node startup/shutdown edge cases that probably aren't worth worrying about right now?) but also worth noting that I'm still feeling my way around the Saturn codebase so might be missing something!

This definitely seems worth shipping now in any case - we can refine as we go!

File: voyager.js [hunk 2]
```diff
@@ -45,31 +45,64 @@ new BasicTracerProvider().register()
...
-        Zinnia.jobCompleted()
     }
     console.log('Sleeping for 60s...')
     setTimeout(runSaturnBenchmarkInterval, 1000 * 60)
 }
 
+async function getPublicIPv4Address () {
+    // TODO Replace with Voyager API once available
+    const res = await fetch(`https://api.filspark.com/inspect-request`)
+    const { cloudfareAddr: ip } = await res.json()
```
Comment (travis) [inline voyager.js]: should this be `cloudflareAddr`? (missing an `l`)

Comment (travis) [reply voyager.js]: oh golly I just went and looked and no, the `l` is missing in the response to `GET https://api.filspark.com/inspect-request`, lol!

**Review by travis — APPROVED** [review]

File: voyager.js (hunk 1 above)
Comment (juliangruber) [reply voyager.js:101]: Ok, let's merge if there is pressure to do so, otherwise wait for @hannahhoward's review?

File: voyager.js (hunk 2 above)
Comment (juliangruber) [reply voyager.js]: hahah good catch I wasn't aware!

Comment (juliangruber) [reply voyager.js]: https://github.com/filecoin-station/spark-api/pull/232

File: voyager.js (hunk 1 above)
Comment (travis) [reply voyager.js:101]: I think it's totally fine to wait, this isn't "need it today" urgent as far as I know!

**Review by bajtos — CHANGES_REQUESTED** [review]
There is a lot of code churn here because of indentation changes and the function `runSaturBenchmarkInterval()` is becoming a bit of a spaghetti code.

Would you mind keeping the benchmark implementation inside a smaller function (this will preserve indentation/avoid code churn) and move the saturn-check to a new function?

```js
export async function runSaturnBenchmark() {
  // original code from runSaturnBenchmarkInterval 
}

export async function runSaturnBenchmarkInterval() {
  // add the check for running on a Saturn node
  // call runSaturnBenchmark when not running on a Saturn node
}
```

**Review by bajtos — APPROVED** [review]
Much better! 👏🏻 

It would be nice to add some automated tests to verify the code works as intended or at least document the steps for verifying this feature manually.

**Comment by juliangruber** [comment]
Tests will be added once Voyager settles. Atm it's still too much of a moving target

File: voyager.js (hunk 1 above)
Comment (hannahhoward) [reply voyager.js:101]: have we tested this against an actual saturn node to verify it accepts an HTTP connection at root?

Comment (hannahhoward) [reply voyager.js:101]: If this doesn't end up working, you can definitely use `/ipfs` with the health check cid -- https://github.com/filecoin-saturn/orchestrator/blob/651e34809ac02cf174f6fa9444524b8624132b1c/src/cron/health-check.js#L26

**Review by hannahhoward — APPROVED** [review]
LGTM

File: voyager.js (hunk 1 above)
Comment (juliangruber) [reply voyager.js:101]: I've tested this with an actual Saturn node, and it worked 👍

Comment (juliangruber) [reply voyager.js:101]: On the root address, it will redirect to saturn.ms, so it's a good http response code

**Pull request merged**